bundles/org.openhab.binding.openwebnet/src/main/java/org/openhab/binding/openwebnet/internal/handler/OpenWebNetLightingGroupHandler.java

```diff
@@ -0,0 +1,210 @@
+
+    public static final Set<ThingTypeUID> SUPPORTED_THING_TYPES = OpenWebNetBindingConstants.LIGHTING_GROUP_SUPPORTED_THING_TYPES;
+
+    protected Set<String> listOn = new HashSet<String>();
+
+    public OpenWebNetLightingGroupHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    protected void requestChannelState(ChannelUID channel) {
+        super.requestChannelState(channel);
+        Where w = deviceWhere;
+        if (w != null) {
+            try {
+                send(Lighting.requestStatus(w.value()));
+            } catch (OWNException e) {
+                logger.debug("Exception while requesting state for channel {}: {} ", channel, e.getMessage());
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
+            }
+        }
+    }
+
+    @Override
+    protected void refreshDevice(boolean refreshAll) {
+        if (refreshAll) {
+            logger.debug("--- refreshDevice() : refreshing GENERAL... ({})", thing.getUID());
+            try {
+                send(Lighting.requestStatus(WhereLightAutom.GENERAL.value()));
+            } catch (OWNException e) {
+                logger.warn("Excpetion while requesting all devices refresh: {}", e.getMessage());
+            }
+        } else {
+            logger.debug("--- refreshDevice() : refreshing SINGLE... ({})", thing.getUID());
+
+            requestChannelState(new ChannelUID(thing.getUID(), CHANNEL_SWITCH_01));
+        }
+    }
+
+    @Override
+    protected void handleChannelCommand(ChannelUID channel, Command command) {
+        switch (channel.getId()) {
+            case CHANNEL_SWITCH:
+                handleSwitchCommand(channel, command);
+                break;
+            default: {
+                logger.warn("Unsupported ChannelUID {}", channel);
+            }
+        }
+    }
+
+    /**
+     * Handles Lighting switch command for a channel
+     *
+     * @param channel the channel
+     * @param command the command
+     */
+    private void handleSwitchCommand(ChannelUID channel, Command command) {
+        logger.debug("handleSwitchCommand() (command={} - channel={})", command, channel);
+        Where w = deviceWhere;
+        if (command instanceof OnOffType && w != null) {
+            try {
+                if (OnOffType.ON.equals(command)) {
+                    send(Lighting.requestTurnOn(w.value()));
+                } else if (OnOffType.OFF.equals(command)) {
+                    send(Lighting.requestTurnOff(w.value()));
+                }
+            } catch (OWNException e) {
+                logger.warn("Exception while processing command {}: {}", command, e.getMessage());
+            }
+        } else {
+            logger.warn("Unsupported command: {}", command);
+        }
+    }
+
+    @Override
+    protected Who getManagedWho() {
+        return Who.LIGHTING;
+    }
+
+    protected void handlePropagatedMessage(Lighting lmsg, @Nullable String oId) {
```

Comment:
Please adjust this method, previous comments abvout meaningfull names and loggin apply here too.